06_create_cluster.sh

```diff
@@ -23,8 +24,14 @@ fi
 if [ ! -d ocp ]; then
     mkdir -p ocp
 
-    # Extract openshift-install from the release image
-    extract_installer "${OPENSHIFT_RELEASE_IMAGE}" ocp/
+    if [ -z "$INSTALL_FROM_GIT" ]; then
+      # Extract openshift-install from the release image
+      extract_installer "${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}" ocp/
+    else
+      # Clone and build the installer from source
+      clone_installer
+      build_installer
```

Comment:
This needs to be optional, too.  Otherwise the `clone_installer` step will create a new branch to build from instead of letting you build a custom installer out of your tree.

Also, now if you set INSTALL_FROM_GIT but not OPENSHIFT_INSTALLER, this is just going to fail, as ocp/bin/kni-install won't exist.

I think `config_example.sh` needs some updates to make the desired workflow more clear here.

Reply:
The developer use case is really just to override `OPENSHIFT_INSTALLER`, after they've built the project themselves. At least that's my workflow when working on the installer.  Developers don't need to use INSTALL_FROM_GIT, that's mostly for CI or testing out of master from a clean slate.

> Also, now if you set INSTALL_FROM_GIT but not OPENSHIFT_INSTALLER, this is just going to fail, as ocp/bin/kni-install won't exist.

build_installer sets the OPENSHIFT_INSTALLER environment variable, it does pass (that's what CI used when testing this PR)

Reply:
I realize it's kind of clumsy to use, will think about it more tomorrow.  Open to ideas about how to deal with the multiple use cases here.